MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -157,8 +157,14 @@ open class RouteProgress: NSObject {
             
             if let segmentCongestionLevels = leg.segmentCongestionLevels, let expectedSegmentTravelTimes = leg.expectedSegmentTravelTimes  {
                 
-                for step in leg.steps.dropFirst() {
-                    let stepCoordinatesCount = maneuverCoordinateIndex + Int(step.coordinateCount) - 1
+                for step in leg.steps {
+                    guard let coordinates = step.coordinates else { continue }
+                    let stepCoordinates = step.maneuverType == .arrive ? Int(step.coordinateCount) : coordinates.dropLast().count
```

Comment:
Perhaps it might be smarter to only remove the last if the count is equal to two?

Reply:
Conceptually, we want to remove the last coordinate if there’s a following step. Since we’re in a for-in loop, there isn’t a convenient way to do that other than to check that the maneuver type is `arrive`. So I think this is fine. It’s worth checking that multi-leg routes still work as expected, though.